### PR TITLE
Add an Exclusion filter to the Command Explorer.

### PR DESCRIPTION
**Pull request by corbob**
## PR Summary
Add an Exclusion filter to the Command Explorer.

Suggested in  #1677

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
- [x] Summarized changes
- NA PR has tests
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -482,6 +482,11 @@
           "default":true,
           "description": "Specifies the visibility of the Command Explorer in the PowerShell Side Bar."
         },
+        "powershell.sideBar.CommandExplorerExcludeFilter": {
+          "type":"string",
```
Comment (TylerLeonhardt) [inline package.json]: I know they said a comma separated list, but we can just use an array, right?

File: src/features/GetCommands.ts [hunk 2]
```diff
@@ -52,6 +52,15 @@ export class GetCommandsFeature implements IFeature {
...
         }
         this.languageClient.sendRequest(GetCommandRequestType, "").then((result) => {
+            const SidebarConfig = vscode.workspace.getConfiguration("powershell.sideBar");
+            const excludeFilter: string = SidebarConfig.CommandExplorerExcludeFilter;
+            if (excludeFilter !== "") {
+                const filters = excludeFilter.split(",");
+                filters.forEach((filter) => {
+                    result = result.filter((command) =>
+                        !command.moduleName.includes(filter));
+                });
```
Comment (TylerLeonhardt) [inline src/features/GetCommands.ts]: I think this forEach block can be shortened:

```ts
result.filter((command) => !filters.indexOf(command.moduleName));
```

File: package.json (hunk 1 above)
Comment (corbob) [reply package.json]: IMHO the user experience of a comma separated string is better than for an array. If we put it in as an array, it creates a link to edit in the settings.js. I have been trying to replicate the `file.exclude` that is contributed by vscode itself. I appears to come from this code: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/files/electron-browser/files.contribution.ts#L192-L216 unfortunately when I replicate it, it still gives an edit in settings.js.

File: src/features/GetCommands.ts (hunk 2 above)
Comment (corbob) [reply src/features/GetCommands.ts]: Almost correct. Just had to test for indexOf returning -1 since that's what it returns when there's nothing there.

**Comment by TylerLeonhardt** [comment]
@corbob do you think that `result` could ever be null here? That's my only concern.

**Comment by corbob** [comment]
> @corbob do you think that `result` could ever be null here? That's my only concern.

I don't believe it should ever happen, but I'll do some modifications of PSES to send back a null and make sure we handle that properly on the off chance null comes back.

**Comment by TylerLeonhardt** [comment]
cool cool. Looks good then!

**Comment by corbob** [comment]
So I did a test, and result comes back as an empty array when get-command just does a write-host:

image

Tested on both PowerShell 5.1 and 6.2.

**Pull request merged**